### DISCUSSION TITLE
Ability to pause and restart a stream at a specific date

### DISCUSSION THREAD

**Discussion by maxdesalle** [sablier-labs/evm-monorepo#1346] [Ideas]
Would be great if users had the ability to both pause a stream and restart it at a specific time. The use case for payroll, for example, would be to stop the stream at the exact time an employee is planned to leave the company, for example. Or if an employee goes on an unpaid vacation, the employer could plan for that by setting the pause date to the first day of the employee's vacation and the restart date on the first day they get back to work from their vacation.

@sablier-labs/engineers

**Comment by PaulRBerg** [comment]
hmmm I think we discussed this feature at various points in the past:

- https://github.com/sablier-labs/flow/discussions/40
- https://github.com/sablier-labs/flow/discussions/22
- https://github.com/sablier-labs/flow/discussions/46

IIRC, we decided to keep it simple and not enable scheduled until users explicitly request this feature. It looks like that has happened already. @maxdesalle could you please share a screenshot of the wording used by the user?

Open to implementing this — we just need to prioritize.

**Reply by smol-ninja** [reply]
This user wants to create a stream with start date in the past. Its not related to scheduling of the transactions that Max proposed. For this, it makes sense to offer this feature. We can offer a separate `createWithStartTime` function for this which simply sets `lastTimeUpdate` to the time specified.

**Reply by PaulRBerg** [reply]
Good point @smol-ninja, I have removed the screenshot from my response and created a GitHub issue for that feature:

https://github.com/sablier-labs/flow/issues/351
